bin/tag

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+$LOAD_PATH.unshift 'lib'
+
+require 'sdr_deploy'
+
+# CLI for tagging git repos
+class Tag < Thor
+  default_task :tag
+
+  option :message,
+         required: false,
+         banner: 'TAG_MESSAGE',
+         desc: 'Message to describe a newly created tag',
+         aliases: '-m'
+
+  option :delete,
+         required: false,
+         type: :boolean,
+         default: false,
+         banner: 'DELETE',
+         desc: 'Delete the tag locally and remotely',
+         aliases: '-d'
+
+  desc 'tag TAG_NAME', 'create or delete a tag named TAG_NAME'
+  def tag(tag_name)
+    RepoUpdater.update(repos: Settings.repositories)
+    if options[:delete]
+      Tagger.delete(tag_name: tag_name)
+    else
+      Tagger.create(tag_name: tag_name, tag_message: options.fetch(:message, 'created by sdr-deploy'))
+    end
+  end
+
+  def self.exit_on_failure?
+    true
+  end
+end
+
+# Allow Thor default task to be run without naming it twice: https://github.com/rails/thor/issues/370#issuecomment-381423134
+ARGV.unshift(Tag.default_task) unless Tag.all_tasks.key?(ARGV[0])
```

Comment:
We don't need this in the other `bin` scripts because the others are not run with a required argument and without options. This is only needed in `bin/tag` to support the simplest case: `bin/tag v1.0.0`. Without this line, that command fails. Thor does not seem to have a better way to support this. 🤷🏻‍♂️